src/Compilers/CSharp/Test/Semantic/Semantics/BindingAsyncTests.cs

```diff
@@ -3257,18 +3254,33 @@ public void BadSpecialByRefVarDeclLocal()
 class Test
 {
     async Task M1(bool truth)
+    {
+        var tr = new TypedReference(); // 1
+        await Task.Factory.StartNew(() => { });
+    }
+
+    async Task M2(bool truth)
     {
         var tr = new TypedReference();
```

Comment:
>var tr = new TypedReference(); [](http://example.com/codeflow?start=8&length=30)

It would be good to add a test with a `ref` return used across `await` in an expression. Something like:
```
M1(ref GiveMeRef(), await M2());
```

Reply:
We already have similar tests it seems, e.g., `ReadonlyParamAsyncSpillIn` but will add exactly the one you suggest (`RefReturnAcrossAwaitExpression`)